src/ValidationError.js

```diff
@@ -640,18 +785,30 @@ class ValidationError extends Error {
     }
 
     if (Array.isArray(schema.type)) {
-      return `${schema.type.map((item) => item).join(' | ')}`;
+      return `${schema.type.join(' | ')}`;
     }
 
     // Fallback for unknown keywords
     /* istanbul ignore next */
     return JSON.stringify(schema, null, 2);
   }
 
+  /**
+   * @param {Schema=} schemaPart
+   * @param {(boolean | Array<string>)=} additionalPath
+   * @param {boolean=} needDot
+   * @returns {string}
+   */
   getSchemaPartText(schemaPart, additionalPath, needDot = false) {
-    if (additionalPath) {
+    if (!schemaPart) {
+      return '';
+    }
+
+    if (Array.isArray(additionalPath)) {
       for (let i = 0; i < additionalPath.length; i++) {
-        const inner = schemaPart[additionalPath[i]];
+        /** @type {Schema} */
```

Comment:
` /** @type {Schema | undefined} */`

on the next line there is guard for `isTruthy`

Reply:
:+1: